crates/bevy_ecs/src/system/commands/mod.rs

```diff
@@ -959,6 +979,27 @@ where
     }
 }
 
+/// A [`Command`] that adds a [`Component`] to an entity.
+#[derive(Debug)]
+pub struct InsertById {
+    /// The entity to which the component will be added.
+    pub entity: Entity,
+    /// The [`ComponentId`] of the component that will be added.
+    pub id: ComponentId,
+    /// The component that will be added.
+    pub component: Box<dyn Component>,
+}
+
+impl Command for InsertById {
+    fn apply(self, world: &mut World) {
+        if let Some(mut entity) = world.get_entity_mut(self.entity) {
+            entity.insert_by_id(self.id, self.component);
```

Comment:
Shouldn't this be inside an `unsafe` block?
I think you need to verify the safety constraints are valid before calling it too.
For `remove_by_id` you can just check that `self.id` is indeed a valid component id, but for `InsertById` you need to verify that `self.component` has the right underlying type and I'm not sure how to do that.

Reply:
I've added a `unsafe` block and implemented the component type validation, but I'm not sure if it's right. Also, the stuff related to `EntityCommands::insert_by_id` method is not right yet, there still a couple of errors